web/packages/teleport/src/Roles/RoleEditor/StandardEditor/Resources.tsx

```diff
@@ -369,6 +371,64 @@ export function KubernetesAccessSection({
   );
 }
 
+function KubernetesResourceKindView({
+  value,
+  validation,
+  isProcessing,
+  onChange,
+  roleVersion,
+}: {
+  value: KubernetesResourceModel;
+  validation: KubernetesResourceValidationResult['kind'];
+  isProcessing: boolean;
+  onChange?(m: KubernetesResourceModel): void;
+  roleVersion: RoleVersion;
+}) {
+  const supportsCRDs = supportsKubernetesCustomResources(roleVersion);
+  const selectField = (
+    <FieldSelect
+      label="Kind"
+      isDisabled={isProcessing}
+      options={kubernetesResourceKindOptions.filter(
+        elem => supportsCRDs || elem.label !== 'Custom'
+      )}
+      value={value.kind}
+      rule={precomputed(validation)}
+      onChange={k => onChange?.({ ...value, kind: k })}
+    />
+  );
+
+  // If we are on version <8, we don't support custom resources. Only show the selectField.
+  if (!supportsKubernetesCustomResources(value.roleVersion)) {
+    return selectField;
+  }
+
+  if (value.kind.label !== 'Custom') {
+    // If we are not dealing with a custom resource, return the select field.
+    return selectField;
+  }
+
+  // If we are dealing with a custom resource, show an input field.
+  return (
+    <>
+      {selectField}
+      <FieldInput
+        label="Custom Kind"
+        disabled={isProcessing}
+        value={value.kind.value}
+        placeholder='e.g. "CronTab"'
+        rule={precomputed(validation)}
+        onChange={e =>
+          onChange?.({
+            ...value,
+            kind: { ...value.kind, value: e.target.value },
```

Comment:
Ooooh, now I understand why you are using the label for making decisions. It's a bit hacky and makes the model hard to understand. I think that if we are using two separate UI fields for this, we should just use two separate model fields.

One other option is to simply replace the entire "add another input for custom" strategy with a `FieldSelectCreatable`, which simply allows the user to type anything. In such case, you'd have a single model field. I'm fine with both strategies.